### PR TITLE
config/v1/types_cluster_version: Add CBO

### PR DESCRIPTION
**Pull request by honza**
Add cluster-baremetal-operator as a capability

### CODE REVIEW

**Comment by honza** [comment]
@wking Would you mind having a look since you added the initial capability code?

File: config/v1/types_cluster_version.go [hunk 1]
```diff
@@ -237,6 +237,13 @@ const (
 	ClusterVersionCapabilityOpenShiftSamples ClusterVersionCapability = "openshift-samples"
 )
 
+const (
```
Comment (wking) [inline config/v1/types_cluster_version.go:240]: nit, no need to close the previous `const (` block just to open a new one here.  You can drop this new line and the `)` before.

File: config/v1/types_cluster_version.go [hunk 2]
```diff
@@ -225,7 +225,7 @@ type UpdateHistory struct {
 type ClusterID string
 
 // ClusterVersionCapability enumerates optional, core cluster components.
-// +kubebuilder:validation:Enum=openshift-samples
+// +kubebuilder:validation:Enum=openshift-samples;cluster-baremetal-operator
```
Comment (wking) [inline config/v1/types_cluster_version.go:228]: After [this thread][1], you moved the constant value to `baremetal`, but you need to update this enum declaration to match.

[1]: https://github.com/openshift/api/pull/1125#discussion_r818158011

File: config/v1/types_cluster_version.go [hunk 3]
```diff
@@ -235,6 +235,11 @@ const (
 	// needed for the image streams to import the images they
 	// reference.
 	ClusterVersionCapabilityOpenShiftSamples ClusterVersionCapability = "openshift-samples"
+
+	// ClusterVersionCapabilityClusterBaremetalOperator manages the cluster
+	// baremetal operator which is responsible for running the metal3
+	// deployment.
+	ClusterVersionCapabilityClusterBaremetalOperator ClusterVersionCapability = "baremetal"
```
Comment (wking) [inline config/v1/types_cluster_version.go:242]: nit: after [this thread][1], you moved the constant value to `baremetal`.  I'm fairly agnostic, but you might want to name the constant `ClusterVersionCapabilityBaremetal` to match.

[1]: https://github.com/openshift/api/pull/1125#discussion_r818158011

**Review by wking — APPROVED** [review]
/lgtm

**Pull request merged**